jgrapht-core/src/main/java/org/jgrapht/alg/treedecomposition/NiceDecompositionBuilder.java

```diff
@@ -0,0 +1,196 @@
+package org.jgrapht.alg.treedecomposition;
+
+import java.util.*;
+
+import org.jgrapht.*;
+import org.jgrapht.alg.util.*;
+import org.jgrapht.graph.*;
+
+/**
+ * A builder for nice tree decompositions. A tree decomposition of a graph G is a tree T and a map
+ * b: V(T) to Set of V(G), which satisfies the properties: - for every edge e in E(G), there is a
+ * node t in V(T) with e is a subset of b(v) - for all vertices v in V(G) the set {t in V(T) | v in
+ * b(t)} is non-empty and connected in T
+ * 
+ * A nice tree decomposition is a special tree decomposition, which satisfies the properties: - for
+ * root r in V(T) and leaf l in V(T): b(r)=b(t)=empty set - every non-leaf node t in V(T) is of one
+ * of the following three types: - introduce node: t has exactly one child d and b(t) = b(d) union w
+ * for some w in V(G) - forget node: t has exactly one child d and b(t) union w = b(d) for some w in
+ * V(G)\b(t) - join node: t has exactly two child d_1, d_2 and b(t)=b(d_1)=b(d_2)
+ * 
+ * @author Ira Justus Fesefeldt (PhoenixIra)
+ *
+ * @param <V> the vertices of G
+ */
+public class NiceDecompositionBuilder<V>
```

Comment:
from my perspective the class name builder is confusing. When the API users would then expect 1) each method building the component returns the builder itself and 2) after building the component the final product can then be retrieved by a build method, the return type of which is Graph<Integer, DefaultEdge> in this case.